src/modal/src/components/Modal.test.js

```diff
@@ -322,6 +322,22 @@ describe('modal', () => {
         expect(mockOnClose).toHaveBeenCalledTimes(1);
       });
 
+      test('should not call close() if the modal is not visible', () => {
+        const mockOnClose = jest.fn();
+        shallow(
+          <Modal isVisible={false} onClose={mockOnClose} shouldCloseOnEscape>
+            { MOCK_CHILD }
+          </Modal>
+        );
+        mockAddEventListenerMap.keydown({ key: 'Escape', code: 'Escape' });
+        mockAddEventListenerMap.keyup({ key: 'Escape', code: 'Escape' });
+        mockAddEventListenerMap.keydown({ key: 'Escape', code: 'Escape' });
+        mockAddEventListenerMap.keyup({ key: 'Escape', code: 'Escape' });
+        mockAddEventListenerMap.keydown({ key: 'Escape', code: 'Escape' });
+        mockAddEventListenerMap.keyup({ key: 'Escape', code: 'Escape' });
```

Comment:
What's all this about?

Reply:
I wanted to simulate hitting the escape key multiple times, cuz sometimes you really wanna slam your escape key.